### ISSUE TITLE
Relevance of Social Providers

### ISSUE THREAD

**Issue by mleanos** [meanjs/mean#1732]
For a while I've been thinking about the relevance of the Social Providers in MEANJS framework. Most of the projects I've worked on don't require this feature at all; due to the CMS nature of the structure of MEANJS. Furthermore, I'm not a fan of the current Social Providers implementation.

I'd like to propose to remove the Social Providers feature and, perhaps at a later time, take another shot at implementing them in a cleaner way that makes sense for MEANJS.

Side Note: The Social Providers implementation is the only thing holding up the [JWT pull-request](https://github.com/meanjs/mean/pull/1300). However, this is not the main reason for me to suggest this. In almost all projects I've started from MEANJS, I had to remove this feature.

**Comment by Wuntenn** [comment]
I've only used this for my main project (product) and a couple experiments. I've found it useful having so many providers to chose from. I kinda agree with you, but I like having the option to get to the start point using the delete key instead of my few remaining brain-buds! :D

There's quite a few hours tucked away into those files by people who are familiar with the whole framework well. If we remove social providers completely I get the feeling that people approaching the project anew and without  awareness of previous implementation would write their own possibly  inferior versions whilst they familiarise themselves with it.

I was thinking along the same lines as you when I suggested that we remove the authentication/users and re-implement them as optional modules. There should be a way to start the project with a minimal set features and add only those that suit your projects requirements.

This would give the project wider use and we'd get to have both JWT and OAuth.

**Comment by adamdilibero** [comment]
I agree with Wuntenn, Not that I have much say around here but I'd prefer JWT sooner & re-implement the social providers as optional modules than delay JWT any further.

**Comment by lirantal** [comment]
@mleanos is there some way to maintain social providers functionality along with JWT? maybe with some refactoring to the way the social provides implementation works?

**Comment by mleanos** [comment]
@lirantal I'm planning on refactoring the social providers features. We should be able to use JWT, and find a suitable solution for implementing Social Providers. I hope to have time within the next week to submit a PR.

@Wuntenn I agree that having the social providers is very useful. However, in my experience with developing CMS type applications from MEANJS, it isn't relevant. For the purpose of this framework, I think it's worthwhile to have it though.

**Comment by simison** [comment]
We're using them at Trustroots.org and it was really nice not to have to code them ourself. That said, might indeed make sense to streamline mean.js a bit and drop them out. Just any community-like project probably finds them useful.
 
>I'm not a fan of the current Social Providers implementation.

Wanna elaborate? :-) Just curious.

**Comment by mleanos** [comment]
@simison The main issues I have is with the logic residing here: https://github.com/meanjs/mean/blob/master/modules/users/server/controllers/users/users.authentication.server.controller.js#L124

It needs some tidying up, and has some bugs.

On the client-side, I'd rather see [this functionality](https://github.com/meanjs/mean/blob/master/modules/users/client/controllers/authentication.client.controller.js#L58-L63) (which also exists [here](https://github.com/meanjs/mean/blob/master/modules/users/client/controllers/settings/manage-social-accounts.client.controller.js#L49-L52)) added to a SocialProviders client-side service. This is pretty minor, but could be a step in the right direction to have a cleaner overall implementation. This would help our users to better understand how this feature is working.
